### PR TITLE
merge task check fix & instances/vms collections updates

### LINKED ISSUE #10: Provider Support for AWS
Have CLI support for the AWS provider.
Support to create, teardown, add networking capabilities, query the resource (vm)

### PR DESCRIPTION
**Pull request by smatula**
## What does this implement/fix? Explain your changes
This is a merge of the check task status fix into the AWS provision examples. The merge of the update to instances/vms collections changes into OSP provision example. Fix checks against active in examples to include also finished. Check may miss active and go straight to finish. 

## Does this close any currently open issues?

Relates to #112 
Closes #10

### CODE REVIEW

File: example/aws_provision_example.py [hunk 1]
```diff
@@ -39,23 +43,82 @@
...
+    # Get the data back from the provision_request
+    client.collection = "provision_requests"
+    result = client.collection.status(req_id)
+    vm_name = payload_data["vm_name"]
+
+    # Set key attributes if provided
+    vm_provider = None
+    vm_network = None
+    vm_tenant = None
+    vm_subnet = None
```
Comment (vi-patel) [inline example/aws_provision_example.py]: I believe subnet is not a value that OpenStack instances have.

File: example/osp_provision_example.py [hunk 2]
```diff
@@ -134,11 +135,43 @@
...
+    # Set key attributes if provided
+    vm_provider = None
+    vm_network = None
+    vm_tenant = None
+    vm_subnet = None
+    if "provider" in payload_data and payload_data["provider"]:
+        vm_provider = payload_data["provider"]
+    if "network" in payload_data and payload_data["network"]:
+        vm_network = payload_data["network"]
+    if "subnet" in payload_data and payload_data["subnet"]:
```
Comment (vi-patel) [inline example/osp_provision_example.py]: I believe subnet is not a value that OpenStack instances have.

**Review by vi-patel — APPROVED** [review]
looks good.

**Review by ryankwilliams — APPROVED** [review]
LGTM

**Pull request merged**